### PR TITLE
Style + refactor/insights

### PR DESCRIPTION
**Pull request by jackcasstlesjones**
Screenshot 2024-12-08 at 14 05 40
Screenshot 2024-12-08 at 14 05 51
Screenshot 2024-12-08 at 14 06 38

# What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Styling
- [ ] Build
- [ ] Chore

## Description

- Changed styling on Insights graph to look more pleasant
- Moved date picker component to `InsightsDisplay.tsx` so that it can be used for all graphs
- Conditionally rendered the x-axis so that it displays different date formats depending on the date range selected

## Screenshots

Screenshot 2024-12-08 at 14 05 40
Screenshot 2024-12-08 at 14 05 51
Screenshot 2024-12-08 at 14 06 38

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: Not go time, happy to write tests before being merged, testing suite seems buggy. There are tests for categories that seem flaky - worth having a look at but might be one for week 4 IMO.
- [ ] I need help with writing tests

## Delete branch after merge?

- [ ] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

(about the graph)

alt_text

### CODE REVIEW

File: src/app/insights/components/InsightsDisplay.tsx
Comment (maxitect) [inline src/app/insights/components/InsightsDisplay.tsx:152]: I think this is not working, but we are getting the loading message from PlotlyChart.tsx, perhaps instead we can pass a prop for customised loading messages if we want this behaviour? [not within the scope of this PR, will not block merge] we could remove the ternary condition within this PR though [will merge if refused]

**Review by maxitect — APPROVED** [review]
Really great work, chart looks super duper! I am approving this PR but not merging until my comment is resolved as it wouldn't block. I will also wait until the backlog of PRs is cleared so you can merge with main before merging this one.

File: src/app/insights/components/InsightsDisplay.tsx
Comment (jackcasstlesjones) [reply src/app/insights/components/InsightsDisplay.tsx:152]: are you talking about the `loading insights...` message? If so, it is working! Just have to watch the screen carefully as it only pops up for a second

Comment (maxitect) [reply src/app/insights/components/InsightsDisplay.tsx:152]: Oh I see my bad yes I see it

**Pull request merged**
